lldb/source/Plugins/Process/AIX/NativeProcessAIX.cpp

```diff
@@ -0,0 +1,281 @@
+  if (m_terminal_fd != -1)
+    cantFail(SetFDFlags(m_terminal_fd, O_NONBLOCK));
+
+  // Let our process instance know the thread has stopped.
+  SetCurrentThreadID(tids[0]);
+  SetState(StateType::eStateStopped, false);
+}
+
+llvm::Expected<std::vector<::pid_t>> NativeProcessAIX::Attach(::pid_t pid) {
+  Log *log = GetLog(POSIXLog::Process);
+  Status status;
+  if (!PtraceWrapper(PT_ATTACH, pid))
+    return errorCodeToError(errnoAsErrorCode());
+
+  int wpid = llvm::sys::RetryAfterSignal(-1, ::waitpid, pid, nullptr, WNOHANG);
+  if (wpid <= 0)
+    return llvm::errorCodeToError(errnoAsErrorCode());
+  LLDB_LOG(log, "adding pid = {0}", pid);
+
+  return std::vector<::pid_t>{pid};
+}
+
+bool NativeProcessAIX::SupportHardwareSingleStepping() const { return false; }
+
+Status NativeProcessAIX::Resume(const ResumeActionList &resume_actions) {
+  return Status("unsupported");
+}
+
+Status NativeProcessAIX::Halt() { return Status("unsupported"); }
+
+Status NativeProcessAIX::Detach() { return Status("unsupported"); }
+
+Status NativeProcessAIX::Signal(int signo) { return Status("unsupported"); }
+
+Status NativeProcessAIX::Interrupt() { return Status("unsupported"); }
+
+Status NativeProcessAIX::Kill() { return Status("unsupported"); }
+
+Status NativeProcessAIX::ReadMemory(lldb::addr_t addr, void *buf, size_t size,
+                                    size_t &bytes_read) {
+  return Status("unsupported");
+}
+
+Status NativeProcessAIX::WriteMemory(lldb::addr_t addr, const void *buf,
+                                     size_t size, size_t &bytes_written) {
+  return Status("unsupported");
+}
+
+size_t NativeProcessAIX::UpdateThreads() {
+  // The NativeProcessAIX monitoring threads are always up to date with
+  // respect to thread state and they keep the thread list populated properly.
+  // All this method needs to do is return the thread count.
+  return m_threads.size();
+}
+
+Status NativeProcessAIX::GetLoadedModuleFileSpec(const char *module_path,
+                                                 FileSpec &file_spec) {
+  return Status("unsupported");
+}
+
+Status NativeProcessAIX::SetBreakpoint(lldb::addr_t addr, uint32_t size,
+                                       bool hardware) {
+  if (hardware)
+    return SetHardwareBreakpoint(addr, size);
+  return SetSoftwareBreakpoint(addr, size);
+}
+
+Status NativeProcessAIX::RemoveBreakpoint(lldb::addr_t addr, bool hardware) {
+  if (hardware)
+    return RemoveHardwareBreakpoint(addr);
+  return NativeProcessProtocol::RemoveBreakpoint(addr);
+}
+
+llvm::Error NativeProcessAIX::Detach(lldb::tid_t tid) {
+  if (PtraceWrapper(PT_DETACH, tid))
+    return llvm::Error::success();
+  return errorCodeToError(errnoAsErrorCode());
+}
+
+llvm::Expected<int> NativeProcessAIX::PtraceWrapper(int req, lldb::pid_t pid,
+                                                    void *addr, void *data,
+                                                    size_t data_size) {
+  long int ret;
```

Comment:
Does ptrace64 `long int` or `int`? If it's `long int` then `PtraceWrapper` should return it as well. Otherwise, this should be an `int`.